src/does-match.js

```diff
@@ -37,45 +37,108 @@
       return string;
     }
 
-    var replaced = '';
-    for (var i = 0, size = string.length; i < size; i++) {
-      replaced += replaceDiacritics(string[i]);
-    }
+    var replaced = string.split('').map(replaceDiacritics);
 
-    return replaced;
+    return replaced.join('');
```

Comment:
I gave it some thought and changed my mind. I'll take a more sensible approach and accept you PR as is and perform some benchmark later, and optimize as needed.